canary.go

```diff
@@ -49,58 +50,100 @@ func (c *Canary) SignalHandler() {
 			}
 			os.Exit(0)
 		case syscall.SIGHUP:
+			manifest, err := manifest.GetManifest(c.Config.ManifestURL, c.Config.DefaultSampleInterval)
+			if err != nil {
+				log.Fatal(err)
+			}
 			// Split reload logic into reloader() as to allow other things to trigger a manifest reload
-			c.ReloadChan <- true
+			c.ReloadChan <- manifest
 		}
 	}
 }
 
 func (c *Canary) reloader() {
 	if c.ReloadChan == nil {
-		c.ReloadChan = make(chan bool)
+		c.ReloadChan = make(chan manifest.Manifest)
 	}
 
-	for r := range c.ReloadChan {
-		if r {
-			// stop all running sensors
-			for _, sensor := range c.Sensors {
-				sensor.Stop()
+	for m := range c.ReloadChan {
+		stoppingSensors := []sensor.Sensor{}
+		for _, sensor := range c.Sensors {
+			found := false
+			for _, newTarget := range m.Targets {
+				if newTarget.Hash == sensor.Target.Hash {
+					found = true
+				}
 			}
-			for _, sensor := range c.Sensors {
-				<-sensor.IsStopped
+			if !found {
+				sensor.Stop()
+				stoppingSensors = append(stoppingSensors, sensor)
 			}
 
-			// get an updated manifest.
-			manifest, err := manifest.GetManifest(c.Config.ManifestURL, c.Config.DefaultSampleInterval)
-			if err != nil {
-				log.Fatal(err)
-			}
-			c.Manifest = manifest
-			if c.Config.RampupSensors {
-				c.Manifest.GenerateRampupDelays(c.Config.DefaultSampleInterval)
-			}
-			// Start new sensors:
-			c.startSensors()
 		}
+		for _, sensor := range stoppingSensors {
```

Comment:
any reason to block here? are you worried about a race somewhere?

if one of the sensors gets wedged, we just block forever?